test/__init__.py

```diff
@@ -642,7 +654,7 @@ def supports_secondary_read_pref(self):
         if self.has_secondaries:
             return True
         if self.is_mongos:
-            shard = self.client.config.shards.find_one()['host']
+            shard = self.client.config.shards.find_one()['host']  # type: ignore[index]
```

Comment:
This is a very common pattern in the test suite. Should we globally ignore the "index" code? Eg:
```python
coll.insert_one({})
coll.find_one()['_id']  # Test can safely assume find_one will never return None here
```

Reply:
Makes sense, updated the rule and removed the now unused comments.